src/scripts/main.js

```diff
@@ -1,3 +1,42 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('tbody');
+const rows = table.querySelectorAll('tr');
+const header = document.querySelector('thead');
+
+header.addEventListener('click', ev => {
+  const cellNum = ev.target.cellIndex;
+  const cellContent = rows[0].children[cellNum].textContent;
+  let cellDataType = '';
+
+  if (Number.isInteger(+cellContent)) {
+    cellDataType = 'num';
+  } else if (cellContent.startsWith('$')) {
+    cellDataType = 'usd';
+  } else {
+    cellDataType = 'str';
+  }
```

Comment:
You can create a separate function to calculate cellDataType to make your code cleaner. 
Now it looks a bit complicated with else...if